src/map/mob.c

```diff
@@ -4719,6 +4712,85 @@ static void mob_drop_ratio_adjust(void){
 }
 
 /**
+ * Copy skill from DB to monster
+ * @param mob Monster DB entry
+ * @param skill Monster skill entries
+ **/
+static void mob_skill_db_set_single_sub(struct mob_db *mob, struct s_mob_skill *skill) {
+	uint8 i;
+
+	nullpo_retv(mob);
+	nullpo_retv(skill);
+
+	for (i = 0; mob->maxskill < MAX_MOBSKILL && i < skill->count; i++) {
+		mob->skill[mob->maxskill++] = skill->skill[i];
+	}
+
+	if (i < skill->count)
```

Comment:
using i here it's correct but weird. it's prone to create an error one day. I think (MAX_MOBSKILL < skill->count) would have been a easier choice.
And beside how could i be superior if ( i < skill->count) is a out condition for the loop above..